stl/inc/yvals_core.h

```diff
@@ -305,13 +305,11 @@
 #ifndef _HAS_CONDITIONAL_EXPLICIT
 #ifdef __cpp_conditional_explicit
 #define _HAS_CONDITIONAL_EXPLICIT 1
-#elif defined(__CUDACC__)
-#define _HAS_CONDITIONAL_EXPLICIT 0 // TRANSITION
-#elif defined(__clang__)
-#define _HAS_CONDITIONAL_EXPLICIT 0 // TRANSITION, LLVM-42694
-#else // vvv C1XX or non-CUDA EDG vvv
+#elif defined(__clang__) || defined(__CUDACC__) || defined(__INTEL_COMPILER)

```

Comment:
Note that this is untested, I got `__INTEL_COMPILER` from the documentation of Intel's C++ compiler. If someone out there has a Windows install of ICC handy and could throw this it at I'd be obliged.

Reply:
I think it's fine given our support level for ICC.